player/command.c

```diff
@@ -7744,6 +7744,19 @@ void mp_option_change_callback(void *ctx, struct m_config_option *co, int flags,
     if (flags & UPDATE_DEMUXER)
         mpctx->demuxer_changed = true;
 
+    if (flags & UPDATE_AD) {
+        uninit_audio_chain(mpctx);
+        reinit_audio_chain(mpctx);
+    }
+
+    if (flags & UPDATE_VD) {
+        struct track *track = mpctx->current_track[0][STREAM_VIDEO];
+        uninit_video_chain(mpctx);
+        reinit_video_chain(mpctx);
+        if (track)
+            queue_seek(mpctx, MPSEEK_RELATIVE, 0.0, MPSEEK_EXACT, 0);
```

Comment:
we need to seek in video case, but not audio?

Reply:
The video visibly skips ahead a few frames if you don't seek and switch. Even while paused. ffmpeg logs a few errors (much more if you don't seek), but that's probably unavoidable given we're killing the decoder in the middle of playback. With audio, I don't hear any serious difference other than the inherent blip from switching decoders in the middle of playback.

Reply:
Ok, I see.